Failure to build on macOS with Python 3.9.0

### What did you do?

I'm a Homebrew maintainer, trying to build Pillow as part of a Homebrew package: https://github.com/Homebrew/homebrew-core/pull/62795 on macOS with Python 3.9.0. This is one example, but other packages using Pillow fail in the same way.

### What did you expect to happen?

Compilation succeeds.

### What actually happened?

```
    The headers or library files could not be found for zlib,
    a required dependency when compiling Pillow from source.
```

### What are your OS, Python and Pillow versions?

* OS: macOS 10.13 to 10.15
* Python: 3.9.0
* Pillow: 7.2.0

Adding `CFLAGS` seems to get the build past this stage. However, I get a later failure with:

```
    src/_imagingft.c:25:10: fatal error: 'ft2build.h' file not found
    #include <ft2build.h>
             ^~~~~~~~~~~~
```

We're never use `freetype` before for this build, and it used to pass. Does Pillow now require freetype? I would expect it to detect it and, if it does not find it, not use it.

--- a/setup.py
+++ b/setup.py
@@ -477,6 +477,18 @@ class pil_build_ext(build_ext):
                 _add_directory(library_dirs, "/usr/X11/lib")
                 _add_directory(include_dirs, "/usr/X11/include")
 
+            # SDK install path
+            try:
+                sdk_path = (
+                    subprocess.check_output(["xcrun", "--show-sdk-path"])
+                    .strip()
+                    .decode("latin1")
+                )
+            except Exception:
+                sdk_path = None
+            if sdk_path:
+                _add_directory(library_dirs, os.path.join(sdk_path, "usr", "lib"))
+                _add_directory(include_dirs, os.path.join(sdk_path, "usr", "include"))
         elif (
             sys.platform.startswith("linux")
             or sys.platform.startswith("gnu")

